dotnet/src/Extensions/Extensions.UnitTests/TemplateEngine/Prompt/BasicPromptTemplateTests.cs

```diff
@@ -379,32 +367,17 @@ private static MethodInfo Method(Delegate method)
         return method.Method;
     }
 
-    private void MockFunctionRunner(ISKFunction function)
+    private SKContext MockContext(params ISKFunction[] functions)
     {
-        this._functionRunner.Setup(r => r.RunAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<ContextVariables>(), It.IsAny<CancellationToken>()))
-            .Returns<string, string, ContextVariables, CancellationToken>(async (pluginName, functionName, variables, cancellationToken) =>
-            {
-                var context = new SKContext(this._functionRunner.Object, this._serviceProvider.Object, this._serviceSelector.Object, variables);
-                return await function.InvokeAsync(context, null, cancellationToken);
-            });
-    }
+        var plugins = new SKPluginCollection();
 
-    private void MockFunctionRunner(List<ISKFunction> functions)
-    {
-        this._functionRunner.Setup(r => r.RunAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<ContextVariables>(), It.IsAny<CancellationToken>()))
-            .Returns<string, string, ContextVariables, CancellationToken>(async (pluginName, functionName, variables, cancellationToken) =>
-            {
-                var context = new SKContext(this._functionRunner.Object, this._serviceProvider.Object, this._serviceSelector.Object, variables);
-                var function = functions.First(f => f.Name == functionName);
-
-                return await function.InvokeAsync(context, null, cancellationToken);
-            });
-    }
+        if (functions?.Length > 0)
+        {
+            plugins.Add(new SKPlugin("plugin", functions));
+        }
```

Comment:
I love how much simpler the tests get when the mocking is replaced by the use of the simple types.